enctests/testframework/utils/outputTemplate.py

```diff
@@ -0,0 +1,102 @@
+      for ref_name, test_info in track.media_references().items():
+          #merge_test_info = {'name': ref_name}
+          #print(ref_name, test_info.metadata)
+          if ref_name == "DEFAULT_MEDIA":
+              #print("Track:", track)
+              continue
+          #if tracks.name not in test_info.metadata[tracks.name][ref_name]:
+          #    print("Not finding:", test_info.metadata[tracks.name][ref_name])
+          #    continue
+          #enc_info = test_info.metadata[tracks.name][ref_name]# [tracks.name]
+          # for key, enc_data in enc_info.items():
+          #     if key == 'results':
+          #         for arg, value in enc_data.items():
+          #             merge_test_info[arg] = value
+          #     else:
+          #         #print("Something ese:", value)
+          #         merge_test_info['encoder_version'] = key
+          #         for arg, value in enc_data.items():
+          #             merge_test_info[arg] = value
+          #print("testInfo:", test_info.metadata)
+          merge_test_info = test_info.metadata['aswf_enctests']['results']
+          merge_test_info['name'] = ref_name
+          results.append(merge_test_info)
+          merge_test_info['media'] = track.name
+          merge_test_info['vmaf_min'] = float(merge_test_info['vmaf']['min'])
+          merge_test_info['vmaf_mean'] = float(merge_test_info['vmaf']['mean'])
+          merge_test_info['vmaf_harmonic_mean'] = float(merge_test_info['vmaf']['harmonic_mean'])
+          merge_test_info['filesize'] = merge_test_info['filesize']
+
+          # We merge the arguments into the dictionary too, as well as merge it into a single string, to make the graphing simpler.
+          args=[]
+          for k,v in test_info.metadata['aswf_enctests']['encode_arguments'].items():
+            args.extend([k,str(v)])
+            merge_test_info[k] = v 
+          merge_test_info['encode_arguments'] = " ".join(args)
+          alltests.append(merge_test_info)
+      tests[track.name] = {'results': results, 'source_info': track.metadata['aswf_enctests']['source_info']}
+
+  for graph in reportconfig.get("graphs", []):
+      df = pd.DataFrame(alltests)
+      df = df.sort_values(by=graph.get("sortby", "name"))
+      #print(df)
+      if graph.get("type", "line") == "bar":
+        fig = px.bar(df, **graph.get("args")) #"x='quality', y='min', color='media', markers=True, text="min")
+      else:
+        fig = px.line(df, **graph.get("args")) #"x='quality', y='min', color='media', markers=True, text="min")
+
+      filename = reportconfig['name']+"-"+graph.get("name")
+      if "directory" in reportconfig:
+        filename = os.path.join(reportconfig['directory'], filename)
+      if os.path.exists(filename):
+        # Running inside docker sometimes doesnt let you write over files. 
+        os.remove(filename)
+      print("Writing out:", filename)
+      fig.write_image(filename)
+      print("Written out:", filename)
```

Comment:
Should we consider putting this in it's own function? For clarity(?)

Reply:
Done.